packages/preview-middleware-client/src/adp/controllers/AddFragment.controller.ts

```diff
@@ -0,0 +1,404 @@
+    /**
+     * JSON Model that has the data
+     */
+    public model: JSONModel;
+    /**
+     * Dialog instance
+     */
+    public dialog: Dialog;
+    /**
+     * Runtime Authoring
+     */
+    public rta: RuntimeAuthoring;
+    /**
+     * Controll Overlays
+     */
+    public overlays: UI5Element[];
+    /**
+     * RTA Command Executor
+     */
+    private commandExecutor: CommandExecutor;
+
+    /**
+     * Initializes controller, fills model with data and opens the dialog
+     */
+    async onInit() {
+        this.model = new JSONModel();
+        this.commandExecutor = new CommandExecutor(this.rta);
+
+        this.dialog = this.byId('addNewFragmentDialog') as unknown as Dialog;
+
+        await this.buildDialogData();
+
+        this.getView()?.setModel(this.model);
+
+        this.dialog.open();
+    }
+
+    /**
+     * Handles the change in aggregations
+     *
+     * @param event Event
+     */
+    onAggregationChanged(event: Event) {
+        let selectedItem = '';
+        const source = event.getSource() as ExtendedEventProvider;
+        if (source.getSelectedItem()) {
+            selectedItem = source.getSelectedItem().getText();
+        }
+        const selectedKey = source.getSelectedKey();
+
+        this.model.setProperty('/selectedAggregation/key', selectedKey);
+        this.model.setProperty('/selectedAggregation/value', selectedItem);
+
+        let newSelectedControlChildren: string[] | number[] = Object.keys(
+            ControlUtils.getControlAggregationByName(this.runtimeControl, selectedItem)
+        );
+
+        newSelectedControlChildren = newSelectedControlChildren.map((key) => {
+            return parseInt(key);
+        });
+
+        const updatedIndexArray: { key: number; value: number }[] = this.fillIndexArray(newSelectedControlChildren);
+
+        this.model.setProperty('/index', updatedIndexArray);
+        this.model.setProperty('/selectedIndex', updatedIndexArray.length - 1);
+    }
+
+    /**
+     * Handles the change in target indexes
+     *
+     * @param event Event
+     */
+    onIndexChanged(event: Event) {
+        const source = event.getSource() as ExtendedEventProvider;
+        const selectedIndex = source.getSelectedItem().getText();
+        this.model.setProperty('/selectedIndex', parseInt(selectedIndex));
+    }
+
+    /**
+     * Handles fragment name input change
+     *
+     * @param event Event
+     */
+    onFragmentNameInputChange(event: Event) {
+        const source = event.getSource() as ExtendedEventProvider;
+        const fragmentName: string = source.getValue().trim();
+        const fragmentList: { fragmentName: string }[] = this.model.getProperty(
+            '/filteredFragmentList/unFilteredFragmentList'
+        );
+
+        const iExistingFileIndex = fragmentList.findIndex((f: { fragmentName: string }) => {
+            return f.fragmentName === `${fragmentName}.fragment.xml`;
+        });
+
+        switch (true) {
```

Comment:
I have to admit that I have never seen `switch/case` being used like that, however, after a quick search I can see why it is nicer than `if/elseif/elseif/elseif`.
Now looking at your conditions, they shouldn't be done in this order and as a flat sequence. You should actually have nested logic like

```ts
if (fragmentName.length <= 0) {
 // case 2
} else {
    // check if the fragmentName is in the list
    if (your_name_check) {
        // case 3 
    } else if (in_the_list) {
       // case 1
    } else {
       // case 4
    }
}
```

Why should you do this: you only loop through the list if the name is valid, and you don't check length only once

Reply:
Changed in: https://github.com/SAP/open-ux-tools/pull/1183/commits/f13674dc7d302b072082058c3dc36078dcc9adc3